modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/**
+ * Checks if element is disabled or whether it exists in a disabled element tree
+ * @param {!Element} element The DOM element to check.
+ * @return {boolean}
+ */
+function isDisabled(node) {
+	return !!(node.disabled || node.closest('[disabled]'));
+}
+
+/**
+ * Listens to the specified event on the given DOM element, but only calls the
+ * given callback listener when it's triggered by elements that match the
+ * given selector or target element.
+ * @param {!Element} element The DOM element the event should be listened on.
+ * @param {string} eventName The name of the event to listen to.
+ * @param {string} selector Css selector that should match the event for the
+ *     listener to be triggered.
+ * @param {!function(!Object)} callback Function to be called when the event
+ *     is triggered. It will receive the normalized event object.
+ * @return {function} Can be used to remove the listener.
+ */
+function delegate(element, eventName, selector, callback) {
+	const eventHandler = (event) => {
+		if (eventName === 'click' && isDisabled(event.target)) {
+			return;
+		}
+
+		if (event.target.matches(selector)) {
+			callback(event);
+		}
+	};
+
+	element.addEventListener(eventName, eventHandler);
+
+	const removeEvent = () => {
+		element.removeEventListener(eventName, eventHandler);
+	};
+
+	removeEvent.dispose = removeEvent;
+
+	return removeEvent;
+}
+
+export default delegate;
+export {delegate};
```

Comment:
Do not export the same thing twice; I know you're just following the old convention that was used in some Liferay codebases, but we now consider it an anti-pattern. The [linter will prevent this in the future](https://github.com/liferay/liferay-frontend-projects/issues/4).